From c6b7c052ded2aae26e94e83aefd4ff7feced8b6f Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Thu, 27 Aug 2015 16:39:08 +0530
Subject: [PATCH] [EM/GPIO]: Enable/Disable Vconn upon vbus connect/disconnect

Enable/Disable vconn when source mode (vbus) is enable/disable.

Change-Id: Iab33046c41762f5a32983910a8e4f21133885786
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-17333
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
---
 drivers/misc/intel_wcove_gpio.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/misc/intel_wcove_gpio.c b/drivers/misc/intel_wcove_gpio.c
index 5675bfd9899f..c9eb008ef2a5 100644
--- a/drivers/misc/intel_wcove_gpio.c
+++ b/drivers/misc/intel_wcove_gpio.c
@@ -164,6 +164,10 @@ static void wcgpio_ctrl_worker(struct work_struct *work)
 		gpiod_set_value_cansleep(info->gpio_otg,
 						evt->is_src_connected);
 
+		/* enable/disable vconn based on the provider(source) event */
+		gpiod_set_value_cansleep(info->gpio_vconn,
+						evt->is_src_connected);
+
 		/* FIXME: vchrgin GPIO is not setting here to select
 		 * Wireless Charging */
 		list_del(&evt->node);
-- 
2.6.0.rc0.2.g7662973

